bayeux.go

```diff
@@ -238,39 +246,26 @@ func (b *Bayeux) connect(out chan TriggerEvent) chan TriggerEvent {
 	return out
 }
 
-func GetSalesforceCredentials() Credentials {
-	clientID := mustGetEnv("SALESFORCE_CONSUMER_KEY")
-	clientSecret := mustGetEnv("SALESFORCE_CONSUMER_SECRET")
-	username := mustGetEnv("SALESFORCE_USER")
-	password := mustGetEnv("SALESFORCE_PASSWORD")
-	tokenURL := mustGetEnv("SALESFORCE_TOKEN_URL")
+func GetSalesforceCredentials(ap AuthenticationParameters) (*Credentials, err) {
 	params := url.Values{"grant_type": {"password"},
-		"client_id":     {clientID},
-		"client_secret": {clientSecret},
-		"username":      {username},
-		"password":      {password}}
-	res, err := http.PostForm(tokenURL, params)
+		"client_id":     {ap.ClientID},
+		"client_secret": {ap.ClientSecret},
+		"username":      {ap.Username},
+		"password":      {ap.Password}}
+	res, err := http.PostForm(ap.TokenURL, params)
 	if err != nil {
 		logger.Fatal(err)
 	}
 	decoder := json.NewDecoder(res.Body)
 	var creds Credentials
 	if err := decoder.Decode(&creds); err == io.EOF {
-		logger.Fatal(err)
+		return nil, err
 	} else if err != nil {
-		logger.Fatal(err)
+		return nil, err
 	} else if creds.AccessToken == "" {
-		logger.Fatalf("Unable to fetch access token. Check credentials in environmental variables")
-	}
-	return creds
-}
-
-func mustGetEnv(s string) string {
-	r := os.Getenv(s)
-	if r == "" {
-		panic(fmt.Sprintf("Could not fetch key %s", s))
+		return nil, fmt.Errorf("Unable to fetch access token: %w", err)
 	}
-	return r
+	return creds, nil
```

Comment:
Did you test this code? `creds` is structure, but the function requires a reference.

Hey @v1v, is there any chance that Robots have any basic CI skeletons to enable the standard Go building process here? We don't need anything sophisticated at the moment.

Reply:
> is there any chance that Robots have any basic CI skeletons to enable the standard Go building process here? 

That's already supported and self-service based. I'll contact you directly and send the URL with those details (it's in a private repo)

Reply:
@mtojek, I have tested this code and resolved the linting issues, please take a look at your convenience.